src/builtins.c

```diff
@@ -1005,7 +1005,9 @@ JL_CALLABLE(jl_f_invoke)
     jl_value_t *argtypes = args[1];
     JL_GC_PUSH1(&argtypes);
     if (jl_is_tuple(args[1])) {
-        // TODO: maybe deprecation warning, better checking
+        jl_depwarn("`invoke(f, (types...), ...)` is deprecated, "
```

Comment:
AFAIK this is the only function that accepts both that is used at runtime. Other ones are generally only used for reflection.

We also stops inferring the tuple of type form for quite some time that's not mentioned anywhere and that's (together with the comment in the invoke implementation) why I think it is meant to be deprecated.